WordPress/Classes/Services/BlogService+Reminders.swift

```diff
@@ -1,15 +1,21 @@
 import Foundation
 
 extension BlogService {
-    @objc func unscheduleBloggingReminders(for blog: Blog) {
-        do {
-            let scheduler = try ReminderScheduleCoordinator()
-            scheduler.schedule(.none, for: blog, completion: { _ in })
-            // We're currently not propagating success / failure here, as it's
-            // it's only used when removing blogs or accounts, and there's
-            // no extra action we can take if it fails anyway.
-        } catch {
-            DDLogError("Could not instantiate the reminders scheduler: \(error.localizedDescription)")
+    @objc func unscheduleBloggingReminders(for blogID: NSManagedObjectID) {
+        let context = ContextManager.shared.mainContext
+        context.performAndWait {
```

Comment:
That's a good point. However, the blogging reminders code we're calling has to run on the main context's thread. The only alternate solution is the one I described in the PR's description, and you can find it here: #20853.

Reply:
@salimbraksa We're accessing the main context from the main queue using `context.performAndWait`. And the objects being accessed or modified all belong to the main context, not the background context. Notice that `unscheduleBloggingReminders` now takes an `NSManagedObjectID`, not an `NSManagedObject`, and the ID is used to retrieve the blog from the main context. 

> If a segment of code, "Code A", performs operations on the main context from a background thread, it could lead to internal state corruption.

This is actually the violation we're trying to fix.

Reply:
@hassaanelgarem I understand; the way you are using the main context is correct. What I'm saying is that, even though it is correct, it could still crash. That's because the main context is a **shared instance,** so if its internal state is corrupted anywhere in the app ( Code A ), it will be corrupted in your code as well ( Code B ), so the app might crash in your code. That's what happened with this [issue](https://a8c.sentry.io/issues/3893472692/).

In an ideal scenario, if the "main context" was properly used throughout the whole codebase, then this code won't crash. But that's not the case right now. 

That's my understanding of how Core Data works, but I could be wrong. 🙏